include/unifex/bulk_schedule.hpp

```diff
@@ -58,11 +61,19 @@ class _schedule_receiver<Integral, Receiver>::type {
             #pragma loop(ivdep)
 #endif
             for (Integral i(0); i < count_; ++i) {
+                if(stop_token.stop_requested()) {
```

Comment:
This breaks vectorisation. I'm open to not interrupting in vectorised form. It's optimistic anyway.

It's still an open question precisely what the contract should be. Should set_next calls be started in order and cancellation only happen before they are started?

Reply:
For the default implementation of bulk_schedule(), which is built on top of schedule(), the handling of cancellation before we start executing should be handled by the underlying schedule(), which would end up just calling set_done() instead of set_value() here.

As a start, we could have code-paths here for the `is_stop_never_possible_v` and `!st.stop_possible()` checks where we hoist the check for the negative case outside the loop.

For the case where we do want to check periodically for a request to stop, we could try a nested loop where the inner loop was vectorised and we only checked for stop once per outer loop iteration.

The number of iterations to be performed by the inner loop could be tuned for the architecture (eg. lane-width for target architecture) or could possibly be tuned by a property/CPO on the receiver. e.g. a `get_batch_size_hint(receiver)` CPO.

Reply:
> For the default implementation of bulk_schedule(), which is built on top of schedule(), the handling of cancellation before we start executing should be handled by the underlying schedule(), which would end up just calling set_done() instead of set_value() here.

Is your point that I don't need a stop_requested() query outside the loop? I'm not doing that here, though, so I'm not quite sure what you're asking for.

Of course, we can do all those optimisations. I would expect a good QOI implementation to do so.

I could hack something in here just to demonstrate a point with the nested loop, to show we don't have to check every iteration. Maybe use the interference size constant as a hint, or just hard code 16 as a fairly reasonable vectorisable size, given that it depends completely on the function anyway. Adding CPOs to do those queries seems a bigger step out of scope of this diff.

I can add the stop_possible queries. That's clearly useful information.